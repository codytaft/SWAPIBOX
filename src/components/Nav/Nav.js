import React, { Component } from 'react';
import './Nav.css';
import Button from '../Button/Button.js';

class Nav extends Component {
  constructor(props) {
    super(props);
  };

  handleClick = event => {
    const { value } = event.target;
    if (value === 'People') {
      this.props.getPeopleData();
    } else if (value === 'Vehicles') {
      this.props.getVehicleData();
    } else if (value === 'Planets') {
      this.props.getPlanetData();
    }
  };

  render() {
    return (
      <div className="nav-wrapper">
        <div className="header-row">
          <h1 className="title-logo">SWAPI-Box</h1>
          <span className="fav-row">
            <Button onClick={this.props.toggleFavorites} name="FAVORITES" />
            <div className="fav-counter">{this.props.favoriteCount}</div>
          </span>
        </div>
        <div className="button-row">
          <Button onClick={this.handleClick} name="People" />
          <Button onClick={this.handleClick} name="Vehicles" />
          <Button onClick={this.handleClick} name="Planets" />
        </div>
      </div>
    );
  };
};

export default Nav;
