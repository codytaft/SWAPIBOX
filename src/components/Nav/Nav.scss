@import "../../styles/mixins.scss";
@import "../../styles/variables.scss";

.nav-wrapper {
  background-color: $color-four;
  border-bottom: 2px solid $color-gold;
  height: 13em;
  margin: auto 0;
  width: 100%;
  z-index: 1; position:relative;
}

.header-row {
  @include flexBetween();
  align-items: center;
  margin-left: 45%;
}

.button-row {
  @include flexBetween();
  align-items: center;
  margin-left: 39%;
  width: 25em;
}

.title-logo {
  border-bottom: 3px solid $color-gold;
  color: $color-one;
  font-family: $font-fam;
  font-size: 3em;
  margin-top: .4em;
  padding: 6px;
  text-shadow: 2px 2px $color-three;
}

.fav-counter {
  align-items: center;
  border: 3px solid $color-gold;
  color: $color-one;
  display: flex;
  justify-content: center;
  height: 2.6em;
  width: 1.9em;
}

.fav-row {
  align-items: center;
  display: flex;
  justify-content: center;
  margin: auto 0;
  padding: 0;
  width: 30em;
}