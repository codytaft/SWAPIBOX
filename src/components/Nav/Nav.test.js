import React from 'react';
import { shallow } from 'enzyme';
import Nav from '../Nav/Nav';

describe('Nav', () => {
  let wrapper;
  let mockGetPeopleData;
  let mockGetVehicleData;
  let mockGetPlanetData;

  beforeEach(() => {
    wrapper = shallow(
      <Nav 
        getPeopleData={mockGetPeopleData}
        getVehicleData={mockGetVehicleData}
        getPlanetData={mockGetPlanetData}
      />
    );
    
    mockGetPeopleData = jest.fn();
    mockGetVehicleData = jest.fn();
    mockGetPlanetData = jest.fn();
  });

  it('Should match snapshot', () => {
    
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should invoke toggleFavorites on click favorites', () => {
    const mockToggleFavorites = jest.fn();
    wrapper = shallow(<Nav toggleFavorites={mockToggleFavorites} />);
   
    wrapper.instance().forceUpdate();
    wrapper.find('[name="FAVORITES"]').simulate('click');

    expect(mockToggleFavorites).toHaveBeenCalled();
  });

  describe('handleClick', () => {
    it('Should invoke getPeopleData when value is people', () => {
      const event = {target: {value: 'People'}};
      const spy = jest.spyOn(wrapper.instance(), "handleClick");
      wrapper.instance().forceUpdate();
      wrapper.find('[name="People"]').simulate('click', event);

      expect(spy).toHaveBeenCalled();
    });

    it('Should invoke getVehicleData when value is Vehicles', () => {
      const event = {target: {value: 'Vehicles'}};
      const spy = jest.spyOn(wrapper.instance(), "handleClick");
      wrapper.instance().forceUpdate();
      wrapper.find('[name="Vehicles"]').simulate('click', event);

      expect(spy).toHaveBeenCalled();
    });

    it('Should invoke getPlanetData when value is Vehicles', () => {
      const event = {target: {value: 'Planets'}};
      const spy = jest.spyOn(wrapper.instance(), "handleClick");
      wrapper.instance().forceUpdate();
      wrapper.find('[name="Planets"]').simulate('click', event);

      expect(spy).toHaveBeenCalled();
    });
    
  });
});
