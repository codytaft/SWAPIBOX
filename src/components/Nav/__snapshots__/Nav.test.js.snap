// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Should match snapshot 1`] = `"<div class=\\"nav-wrapper\\"><div class=\\"header-row\\"><h1 class=\\"title-logo\\">SWAPI-Box</h1><span class=\\"fav-row\\"><button class=\\"FAVORITES\\" value=\\"FAVORITES\\">FAVORITES</button><div class=\\"fav-counter\\"></div></span></div><div class=\\"button-row\\"><button class=\\"People\\" value=\\"People\\">People</button><button class=\\"Vehicles\\" value=\\"Vehicles\\">Vehicles</button><button class=\\"Planets\\" value=\\"Planets\\">Planets</button></div></div>"`;
